src/main/java/randoop/generation/CoverageTracker.java

```diff
@@ -0,0 +1,200 @@
+ */
+public class CoverageTracker {
+  public static CoverageTracker instance = new CoverageTracker();
+
+  private final IRuntime runtime = new LoggerRuntime();
+  private final MemoryClassLoader memoryClassLoader;
+
+  private final ExecutionDataStore executionData = new ExecutionDataStore();
+  private final SessionInfoStore sessionInfos = new SessionInfoStore();
+
+  private final Instrumenter instrumenter;
+  private final RuntimeData data;
+
+  /** Map from method name to coverage details. */
+  private final Map<String, CoverageDetails> coverageDetailsMap = new HashMap<>();
+
+  /** Set of names of all the classes under test */
+  private Set<String> classesUnderTest = new HashSet<>();
+
+  /**
+   * Coverage details related to a single method under test. Tracks total number of branches and
+   * number of uncovered branches.
+   */
+  public static class CoverageDetails {
+    /** Total number of branches. */
+    public int numBranches;
+    /** Number of uncovered branches. */
+    public int uncoveredBranches;
+  }
+
+  private CoverageTracker() {
+    this.memoryClassLoader = new MemoryClassLoader(this);
+    this.instrumenter = new Instrumenter(runtime);
+    this.data = new RuntimeData();
+
+    try {
+      this.runtime.startup(data);
+    } catch (Exception e) {
+      e.printStackTrace();
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Copy the set of names of classes under test.
+   *
+   * @param classesUnderTest names of all the classes under test.
+   */
+  public void setClassesUnderTest(Set<String> classesUnderTest) {
+    this.classesUnderTest = new HashSet<>(classesUnderTest);
+  }
+
+  /**
+   * Attempts to instrument a class and return the byte array representation.
+   *
+   * @param className name of the class to instrument.
+   * @return the instrumented bytes of the class, null if instrumentation failed.
+   */
+  public byte[] instrumentClass(String className) {
+    // Only instrument classes under test and their nested classes.
+    if (!classesUnderTest.contains(className)) {
+      boolean isNestedClass = false;
+
+      // Determine if this class is a nested class, in which case, the name of some
+      // existing class under test would be the prefix of this class's name.
+      for (String classUnderTest : classesUnderTest) {
+        if (className.startsWith(classUnderTest)) {
+          isNestedClass = true;
+          break;
+        }
+      }
+
+      // If it's not explicitly a class under test, or a nested class of one, don't instrument the class.
+      // Examples include java.lang classes.
+      if (!isNestedClass) {
+        return null;
+      }
+    }
+
+    final byte[] instrumented;
+    final String resource = '/' + className.replace('.', '/') + ".class";
+    InputStream original = getClass().getResourceAsStream(resource);
+
+    if (original == null) {
+      System.err.println("No resource with name: " + resource + " found!");
```

Comment:
Can you be more explicit?
Is this a guaranteed precursor to a failure later?
Whose fault is it, and how could it be corrected?
Should Randoop halt?  Randoop is so noisy that the user is likely to overlook this output.

Reply:
The only place where `instrumentClass` in `CoverageTracker` is invoked is from `MemoryClassLoader`'s `loadClass` method where it checks the returned result.
```
final byte[] bytes = coverageTracker.instrumentClass(name);
if (bytes != null) {
  loadedClass = defineClass(name, bytes, 0, bytes.length);
} else {
  loadedClass = super.loadClass(name, resolve);
}
```
In this case, the parent ClassLoader would attempt to load the class.
However, reaching this point would mean MemoryClassLoader wasn't able to find a class that was explicitly under test and thus the parent ClassLoader probably won't find it either. 
So throwing an error in CoverageTracker seems to be the right thing to do and to halt Randoop.

Reply:
OK.  So it sounds like you will change the `println` to `throw new Error`?